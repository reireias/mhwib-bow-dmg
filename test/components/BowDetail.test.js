import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import bows from '@/constants/bow'
import target from '@/components/BowDetail'
import { createStore } from '@/test/helper'

Vue.use(Vuetify)
const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(Vuetify)

describe('instance', () => {
  let store
  beforeEach(() => {
    store = createStore()
  })

  it('should be vue instance', () => {
    const vuetify = new Vuetify()
    const propsData = { bow: bows[0] }
    const wrapper = shallowMount(target, {
      localVue,
      store,
      vuetify,
      propsData,
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
