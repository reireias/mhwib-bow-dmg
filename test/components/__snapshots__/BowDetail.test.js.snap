// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instance should be vue instance 1`] = `
"<div>
  銀火竜の烈弓 [ 312,
  <span class=\\"fire\\">火330</span>
  , 20%, ① ]
</div>"
`;
